### ISSUE TITLE
PercolatorAdapter multiple hits per spectrum 

### ISSUE THREAD

**Issue by oliveralka** [OpenMS/OpenMS#4090]
If search engine annotated multiple hits per spectrum with the same score (e.g. MSGF+). Percolator seems to correctly calculate values, but all PEPs are 1. 

Strangely this does only happens for some files. 

If IDFilter - best - n_peptide_hits it set to 1 - seems to fix the issue.

**Comment by jpfeuffer** [comment]
Hmm can you run the adapter in debug mode to see the percolator output?
Percolator also has an option to just take the best hit per spectrum (but I guess prefiltering is also a good idea if you want to do so).

**Comment by timosachsenberg** [comment]
Best hit per spectrum is in a newer version only

**Comment by jpfeuffer** [comment]
Any updates? This seems serious but not sure if reproducible.

**Comment by timosachsenberg** [comment]
I think it is a known percolator issue.

**Comment by jpfeuffer** [comment]
Ok but I guess our version is now new enough to include the flag right? So solved?

**Comment by jpfeuffer** [comment]
Ping @timosachsenberg @oliveralka

**Comment by jpfeuffer** [comment]
Ping @timosachsenberg

**Comment by timosachsenberg** [comment]
Do we have a test file. Not sure how to easily reproduce this issue.

**Comment by jpfeuffer** [comment]
One could just use our MSGF test files and activate num hits > 1 if not already activated. Then run Percolator and check the output

**Comment by timosachsenberg** [comment]
with 3 spectra? not gonna work

**Comment by timosachsenberg** [comment]
I will do a big search ...

**Comment by jpfeuffer** [comment]
yes, any data should be fine. you can use plfq test data or something. Either it shows up or not. Then we close it as unconfirmed.

**Comment by timosachsenberg** [comment]
I did an MSGF+ search on the iPRG 2015 dataset retaining top 10 hits.
Percolator after PSMFeatureExtractor with train best positive option.

Ok so first some general observation:
- all decoys get a PEP of 1.0 which I think is fine
```
        <UserParam type="float" name="MS:1001492" value="-0.509628"/>
        <UserParam type="float" name="MS:1001491" value="0.520355"/>
        <UserParam type="float" name="MS:1001493" value="1.0"/>
```
And also most of the worse scoring hits have PEP=1.0.
I then kept only the top target hit with q < 0.1.
Of those, none had a PEP of 1.0
Closing as unconfirmed.
